### PR TITLE
addition of pyrolysine to the amino acid list

### PR DESCRIPTION
**Pull request by trishorts**
From the Wiki:

Pyrrolysine (symbol Pyl or O;[1] encoded by the 'amber' stop codon UAG) is an α-amino acid that is used in the biosynthesis of proteins in some methanogenic archaea and bacteria;[2][3] it is not present in humans. It contains an α-amino group (which is in the protonated –NH+
3 form under biological conditions), a carboxylic acid group (which is in the deprotonated –COO− form under biological conditions). Its pyrroline side-chain is similar to that of lysine in being basic and positively charged at neutral pH.

### CODE REVIEW

File: Proteomics/AminoAcidPolymer/Residue.cs [hunk 1]
```diff
@@ -47,6 +47,7 @@ static Residue()
             {"Methionine",     new Residue("Methionine", 'M', "Met",ChemicalFormula.ParseFormula("C5H9NOS"),  ModificationSites.M)},
             {"Phenylalanine",  new Residue("Phenylalanine", 'F', "Phe",ChemicalFormula.ParseFormula("C9H9NO"),  ModificationSites.F)},
             {"Proline",        new Residue("Proline", 'P', "Pro",ChemicalFormula.ParseFormula("C5H7NO"),  ModificationSites.P)},
+            {"Pyrolysine",     new Residue("Pyrolysine", 'O', "Pyl",ChemicalFormula.ParseFormula("C12H19N3O2"),  ModificationSites.P)},
```
Comment (zrolfs) [inline Proteomics/AminoAcidPolymer/Residue.cs]: Pyrrolysine

Comment (trishorts) [reply Proteomics/AminoAcidPolymer/Residue.cs]: lol

**Comment by zrolfs** [comment]
Really thought we were already looking for this. Thanks for checking and fixing it.

File: Test/TestPeptides.cs [hunk 2]
```diff
@@ -777,6 +777,20 @@ public void BadSeqeunce()
             .EqualTo("Couldn't find the closing ] for a modification in this sequence: A["));
         }
 
+        [Test]
+        public static void TestPyrolysine()
```
Comment (zrolfs) [inline Test/TestPeptides.cs]: r

Comment (trishorts) [reply Test/TestPeptides.cs]: RRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRR

**Pull request merged**